dll/win32/shlwapi/utils.cpp

```diff
@@ -0,0 +1,78 @@
+/*
+ * PROJECT:     ReactOS Shell
+ * LICENSE:     LGPL-2.0-or-later (https://spdx.org/licenses/LGPL-2.0-or-later)
+ * PURPOSE:     Implement shell light-weight utility functions
+ * COPYRIGHT:   Copyright 2023 Katayama Hirofumi MZ <katayama.hirofumi.mz@gmail.com>
+ */
+
+#define _ATL_NO_EXCEPTIONS
+#include "precomp.h"
+#include <shellapi.h>
+#include <shlwapi.h>
+#include <shlwapi_undoc.h>
+#include <shlobj_undoc.h>
+#include <shlguid_undoc.h>
+#include <atlstr.h>         // for CStringW
+#include <strsafe.h>        // for StringC... functions
+
+WINE_DEFAULT_DEBUG_CHANNEL(shell);
+
+/*************************************************************************
+ * IContextMenu_Invoke [SHLWAPI.207]
+ *
+ * Used by Win:SHELL32!CISFBand::_TrySimpleInvoke.
+ */
+EXTERN_C
+BOOL WINAPI
+IContextMenu_Invoke(
+    _In_ IContextMenu *pContextMenu,
+    _In_ HWND hwnd,
+    _In_ LPCSTR lpVerb,
+    _In_ UINT uFlags)
+{
+    CMINVOKECOMMANDINFO info;
+    BOOL ret = FALSE;
+    INT iDefItem = 0;
+    HMENU hMenu = NULL;
+    HCURSOR hOldCursor;
+
+    TRACE("(%p, %p, %s, %u)\n", pContextMenu, hwnd, debugstr_a(lpVerb), uFlags);
+
+    if (!pContextMenu)
+        return FALSE;
+
+    hOldCursor = SetCursor(LoadCursorW(NULL, (LPCWSTR)IDC_WAIT));
+
+    ZeroMemory(&info, sizeof(info));
+    info.cbSize = sizeof(CMINVOKECOMMANDINFO);
+    info.hwnd = hwnd;
+    info.nShow = SW_NORMAL;
+    info.lpVerb = lpVerb;
+
+    if (IS_INTRESOURCE(lpVerb))
+    {
+        hMenu = CreatePopupMenu();
+        if (hMenu)
+        {
+            pContextMenu->QueryContextMenu(hMenu, 0, 1, 0x7FFF, uFlags | CMF_DEFAULTONLY);
+            iDefItem = GetMenuDefaultItem(hMenu, 0, 0);
+            if (iDefItem != -1)
+                info.lpVerb = MAKEINTRESOURCEA(iDefItem - 1);
+        }
```

Comment:
Random thought: the code of this function vaguely looks similar (but not identical) to the implementation of SHInvokeCommand from here: https://git.reactos.org/?p=reactos.git;a=blob;f=dll/win32/shlwapi/ordinal.c;hb=6d7aaaade0a4735165e8307d0d7dec3eaedf8156#l3523
Could you check whether on Windows, one of these functions use the other one for its implementation? or they look similar "just by chance" ?

Reply:
> Could you check whether on Windows, one of these functions use the other one for its implementation?

Checked. No dependencies for each other.